src/handlers/assign.rs

```diff
@@ -799,44 +799,86 @@ async fn find_reviewer_from_names(
         .to_string())
 }
 
+#[derive(Eq, PartialEq, Hash)]
+enum ReviewerCandidate {
+    /// This reviewer was directly requested for a review.
+    Direct(String),
+    /// This reviewer was expanded from a team or an assign group.
+    Expanded(String),
+}
```

Comment:
I find this enum to be highly confusing, the origin of the candidate reviewer is a property of the candidate not a variants of the candidate.

I would expect something like this instead:
```rust
struct ReviewerCandidate {
    name: String,
    origin: ReviewerCandidateOrigin,
}

#[derive(Eq, PartialEq, Hash)]
enum ReviewerCandidateOrigin {
    /// This reviewer was directly requested for a review.
    Direct,
    /// This reviewer was expanded from a team or an assign group.
    Expanded,
}
```

Reply:
Maybe my brain is weird. :smile: Thanks.